File: back/src/main/kotlin/br/ufsc/mariaeduardamelohang/servidormatriculaufscpwa/model/Disciplina.java
package br.ufsc.mariaeduardamelohang.servidormatriculaufscpwa.model;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "TB_DISCIPLINA")
public class Disciplina implements Serializable {

	private static final long serialVersionUID = 7229281137744422191L;

	@Id
	String codigo;

	@Column(nullable = false)
	String nome;

	@Column(nullable = false)
	int cargaHoraria;
}

File: back/src/main/kotlin/br/ufsc/mariaeduardamelohang/servidormatriculaufscpwa/model/Disciplina.kt
package br.ufsc.mariaeduardamelohang.servidormatriculaufscpwa.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.IDENTITY
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "TB_DISCIPLINA")
data class Disciplina(
    @Id @GeneratedValue(strategy = IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    val nome: String,

    @Column(nullable = false)
    var cargaHoraria: Int
)